pandas/core/internals/base.py

```diff
@@ -128,6 +135,8 @@ def equals(self, other: object) -> bool:
         if not all(ax1.equals(ax2) for ax1, ax2 in zip(self_axes, other_axes)):
             return False
 
+        if isinstance(self, BlockManager) and isinstance(other, ArrayManager):
```

Comment:
can you make this `if type(self) is not type(other)` and do the mgr_to_mgr import inside this block?

what about the reversed case of `isinstance(self, ArrayManager)` and isinstance(other, BlockManager)`?